src/dict.c

```diff
@@ -820,9 +820,9 @@ void dictSetKey(dict *d, dictEntry* de, void *key) {
         de->key = key;
 }
 
-void dictSetVal(dict *d, dictEntry *de, void *val) {
+void dictSetVal(dictEntry *de, void *val) {
```

Comment:
```suggestion
void dictSetVal(dict *d, dictEntry *de, void *val) {
UNUSED(d);
```
Although a little silly, this makes the actual scope of the change much smaller. If we end up reverting it, we'll preserve a lot more history.

Reply:
Will need to inspect the assembly but,  this might force the compile to put "d" on the stack for nothing. 
It might force higher function that want to setVal to fetch the dict for nothing both in code lines and in assembly instruction (i.e. I am not sure the compiler will be able to optimize it out)

think about this and if you still prefer me to take your suggestion, say so and I will comply.

Reply:
Well whats the verdict `dict *d` in for fun or out?

Reply:
I would rather keep it in for fun. My understanding is as long as we have spare registers and slots it's basically "free" to put another variable on the stack.